### ISSUE TITLE
[Bug]: pushing anything deletes everything on iOS

### ISSUE THREAD

**Issue by 01abhishekjain** [denolehov/obsidian-git#455]
## tl;dr

`git pull` works. But after pushing any kind of change (a new file or edits in an exiting file, etc) simply clears out the entire repository.

## Details
This is how I set up sync for a brand new vault:
1. created a new empty repo on git desktop with a readme file.
2. created an access token. I tried both forms of personal access token (classic and fine grained).
3. installed obsidian on my iPhone (latest iOS) and created a new empty vault (no iCloud sync)
4. installed this plugin and provided my username and the above token
5. used the "clone existing repo" option to clone the above repo (selected "no" on the .obsidian folder prompt)
6. upon restart, the app shows the readme file
7. any changes made to the readme file *from the github desktop* is reflected in app upon "pull"

So far so good.

Now, the easiest way to reproduce the issue is to just select "create backup" option which does work (no error message). However, when I check on github desktop, it shows that the **commit basically just deleted the readme file.**

What other things I tried?
1. I tried with both an empty repo, and a repo which already had the .obsidian folder with few files in it.
2. I tried both fine grained and classic token (in each case, I provided ALL the permissions -- just to be sure)
3. tried initializing a new repo from the app itself.

### Operating system

iOS

### Installation Method

None

### Plugin version

2.17.1

**Comment by Vinzent03** [comment]
So create backup does create a commit with a deleted file? So the issue does not lie in pushing, but in committing? What means 

> commit basically just deleted the readme file.

What exactly happened?

**Comment by turtur1234** [comment]
I also encountered this issue and I noticed that in the source control view all files are always listed under changes & staging/commiting doesn't cause anything to change. My author name & email is set. Is there a way to view the logs on iOS?
![image](https://user-images.githubusercontent.com/25729109/223161444-a66b7fd5-ff1f-42c0-8a3a-f2e33e41eee4.png)

**Comment by Vinzent03** [comment]
@turtur1234 Are you on the latest version 2.17.2?

**Comment by turtur1234** [comment]
Obsidian 1.4.3, Obsidian git 2.17.1, iOS 16.3.1. I got the plugin from inside the app under community plugins, I'm not sure why it's out of date.

**Comment by Vinzent03** [comment]
@turtur1234 Because I updated plugin very recently. Please retry as said on the latest version. I have no issues staging the files individually or committing all at the same time.

**Comment by 01abhishekjain** [comment]
> So create backup does create a commit with a deleted file? So the issue does not lie in pushing, but in committing? What means 
> 
> 
> 
> > commit basically just deleted the readme file.
> 
> 
> 
> What exactly happened?

Yes, creating backup creates (and pushes) a commit which only has file deletions, no matter what the actual diff was.

It is the same behaviour if I simply make an edit in a file and then commit and push.

Does this provide more clarity? I'm happy to create a screen recording if needed.

**Comment by Vinzent03** [comment]
@01abhishekjain May you please try to reproduce it with the latest version 2.17.4? A recording would be great.

**Comment by turtur1234** [comment]
Just checked and 2.17.4 worked for me. @01abhishekjain can you confirm?

**Comment by 01abhishekjain** [comment]
@Vinzent03 @turtur1234 

The issue is resolved in 2.17.4. 

Feel free to close this. Thank you
